cncf_process/general_technical_review.md

```diff
@@ -0,0 +1,502 @@
+- Karmada Instance HA: a highly available managed Karmada control plane can be deployed across multiple management clusters that can span various data centers, thus fulfilling disaster recovery requirements.  
+- Etcd HA: Karmada supports the use of external etcd, enabling the creation of a DR-compliant etcd cluster.
+
+#### Describe the project’s resource requirements, including CPU, Network and Memory.  
+    
+Resource requirements depend on the number of clusters managed by Karmada, as well as the scale of a single cluster. For large-scale clusters, resource requests and limits should be adjusted accordingly.  
+    
+#### Describe the project’s storage requirements, including its use of ephemeral and/or persistent storage.
+
+Karmada control plane does not require persistent or ephemeral storage.
+
+#### Please outline the project’s API Design:  
+##### Describe the project’s API topology and conventions  
+      
+- Karmada exposes a Kubernetes-native declarative API using Custom Resource Definitions (CRDs), mainly include:  
+  - PropagationPolicy/ClusterPropagationPolicy: represents the policy that propagates a group of resources to one or more clusters.  
+  - Cluster: represents the desired state and status of a member cluster.  
+- The API follows Kubernetes conventions for resource specification, status, and metadata, supporting standard CRUD operations via kubectl and the Kubernetes API server.  
+- Karmada supports versioned APIs (e.g., work.karmada.io/v1alpha2) and uses OpenAPI schema validation for resource definitions.   
+- For more details, see the [Karmada API Reference](https://karmada.io/docs/category/karmada-api).
+
+##### Describe the project defaults
+
+- **Security Considerations**  
+  - To avoid the use of insecure algorithms such as 3DES during the communication process, the TLS configuration is set to  `--tls-min-version=VersionTLS13` during the installation of Karmada-related components.  
+- **Policy-related**  
+  - The default replicaSchedulingType of the PropagationPolicy is "Duplicated", which will schedule the workload to all member clusters.   
+- **Installation-related**  
+  * Karmada components and related resources are deployed to the karmada-system namespace by default.
+- **others**
+  - Karmada uses karmada-scheduler to schedule workloads across multi-clusters by default.
+
+##### Outline any additional configurations from default to make reasonable use of the project
+
+- For production, it is recommended to set resource requests and limits based on the cluster scale and expected traffic.  
+- For production, deploy multiple controller replicas.  
+- Set up monitoring and alerting using Prometheus and Grafana for observability.  
+- See the [Karmada configuration](https://karmada.io/docs/administrator/configuration/configure-controllers) for more configuration recommendations.

```

Comment:
The fourth is a good one, but the others seem not to answer the question.
In addition to the controllers, users can change some performance-related flags to match the data scale.